src/features/event_episodes/atoms/episodesResource.ts

```diff
@@ -34,10 +34,14 @@ export const episodesResource = createAsyncAtom(
       );
       if (!responseData) throw 'No data received';
 
-      // Episodes not have any ids
+      // Adapter:
+      // - add missing ids
+      // - set default value for forecasted flag;
       return responseData.map((e) => {
-        if (!e.id) e.id = 'temp_' + nanoid(6);
-        return e;
+        const cp = { ...e };
+        if (!cp.id) cp.id = 'temp_' + nanoid(6);
+        if (!cp.forecasted) cp.forecasted = false;
+        return cp;
```

Comment:
i mean it's okay but takes time to guess what's `cp`, and i still couldn't guess it but i see what it does

Reply:
`cp`[ is common shortcut for unix-like users ](https://linuxize.com/post/how-to-copy-files-and-directories-in-linux/), but if you not familiar with it, I will replace it to `copy`